contrib/core-contract-tests/tests/clar/utils/clarity-parser.ts

```diff
@@ -0,0 +1,346 @@
+    /\(unwrap! \(contract-call\? \.(.+?) (.+?)(( .+?)*)\)/
+  );
+  if (!match) return null;
+
+  const contractName = match[1];
+  const functionName = match[2];
+  const argStrings = splitArgs(match[3]);
+  let fn: any;
+  simnet.getContractsInterfaces().forEach((contract, contractFQN) => {
+    const [_, ctrName] = contractFQN.split(".");
+    if (ctrName === contractName) {
+      fn = contract.functions.find((f) => f.name === functionName);
+      if (!fn) {
+        throw `function ${functionName} not found in contract ${contractName}`;
+      }
+    }
+  });
+  if (!fn) {
+    throw `function ${functionName} not found in contract ${contractName}`;
+  }
+  const args = fn.args.map((arg: any, index: number) =>
+    parseArg(argStrings[index], arg.type)
+  );
+
+  return {
+    contractName,
+    functionName,
+    args,
+  };
+}
+
+/**
+ *
+ * @param arg argument value as string
+ * @param type should be of type ContractInterfaceAtomType
+ * @returns
+ */
+function parseArg(
+  arg: string,
+  type: ContractInterfaceAtomType
+): { type: string; value: ClarityValue } {
+  switch (type) {
+    case "uint128":
+      return { type: "uint", value: Cl.uint(arg.slice(1)) };
+    case "int128":
+      return { type: "int", value: Cl.int(arg) };
+    case "principal":
+      const [address, name] = arg.split(".");
+      return name
+        ? { type: "principal", value: Cl.contractPrincipal(address, name) }
+        : { type: "principal", value: Cl.standardPrincipal(address) };
+    case "bool":
+      return { type: "bool", value: Cl.bool(arg === "true") };
+  }
+  const typeDescriptor = Object.keys(type)[0];
+  switch (typeDescriptor) {
+    case "buffer":
+      const hexValue = arg.toLowerCase().startsWith('0x') ? arg.slice(2) : arg;
+      return { type: "buffer", value: buffer(hexToBytes(hexValue)) };
+    case "string-utf8":
+      return { type: "string", value: Cl.stringUtf8(arg) };
+    case "string-ascii":
+      return { type: "string", value: Cl.stringAscii(arg) };
+    case "tuple":
+      return {
+        type: "tuple",
+        value: parseTuple(
+          arg,
+          (type as { tuple: ContractInterfaceTupleEntryType[] }).tuple
+        ),
+      };
+    case "optional":
+      if (arg === "none") {
+        return {
+          type: "none",
+          value: Cl.none(),
+        };
+      } else {
+        return {
+          type: "some",
+          value: Cl.some(
+            parseArg(
+              arg,
+              (type as { optional: ContractInterfaceAtomType }).optional
+            ).value
+          ),
+        };
+      }
+    default:
+      throw new Error(`Unsupported type ${type}`);
+  }
+}
+
+function extractCallInfo(statement: string) {
+  const match = statement.match(/\(try! \((.+?)\)\)/);
+  if (!match) return null;
+  return { contractName: "", functionName: match[1], args: [] };
+}
+
+export function getContractName(contractId: string) {
+  return contractId.split(".")[1];
+}
```

Comment:
This parser looks somewhat involved.  Is there a way to add a (simple) set of unit tests to verify that all the test parsers work as intended?  Thanks!

Reply:
@friedger @jcnelson happy to do this with fast-check? This way we’ll test and have the ability to also fuzz the interpreter at the same time.

Reply:
Code improvements have been pushed

Reply:
FYI, some prop/fuzz tests I added can be found in https://github.com/stacks-network/stacks-core/pull/4100/commits/3e9b3af440ade34268b7e5c716ad0534c3982759 and https://github.com/stacks-network/stacks-core/pull/4100/commits/34209043ee101bb5923d40108a5f60aeb6f61962.